### PR TITLE
Experimental: Tweak public markets endpoint

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

**Comment by mod** [comment]
@ysv must rebase

File: app/api/v2/entities/market.rb [hunk 1]
```diff
@@ -80,13 +80,53 @@ class Market < Base
           }
         )
 
+        expose(
```
Comment (ysv) [inline app/api/v2/entities/market.rb]: remove

File: app/api/v2/entities/market.rb [hunk 2]
```diff
@@ -80,13 +80,53 @@ class Market < Base
...
+
         expose(
           :state,
           documentation: {
             type: String,
             desc: "Market state defines if user can see/trade on current market."
           }
         )
+
+        expose(
```
Comment (ysv) [inline app/api/v2/entities/market.rb]: remove

File: app/api/v2/entities/market.rb [hunk 3]
```diff
@@ -80,13 +80,53 @@ class Market < Base
...
+          :ticker,
+          using: API::V2::Entities::Ticker::TickerEntry,
+          if: ->(_market, options) do
+            options[:extended]
+          end
+        )
+
+        expose(
```
Comment (ysv) [inline app/api/v2/entities/market.rb]: remove

File: app/api/v2/entities/market.rb [hunk 4]
```diff
@@ -80,13 +80,53 @@ class Market < Base
...
+            type: String,
+            desc: "Market Base Currency."
+          },
+          if: ->(_market, options) do
+            options[:extended]
+          end
+        )
+
+        expose(
+          :quote,
```
Comment (ysv) [inline app/api/v2/entities/market.rb]: remove

File: app/api/v2/public/markets.rb [hunk 5]
```diff
@@ -16,9 +16,52 @@ class OrderBook < Struct.new(:asks, :bids); end
             success: API::V2::Entities::Market
           params do
             use :pagination
+            optional :ordering,
+                     values: { value: %w(asc desc), message: 'public.markets.invalid_ordering' },
+                     default: 'asc',
+                     desc: 'If set, returned values will be sorted in specific order, defaults to \'asc\'.'
+            optional :order_by,
```
Comment (ysv) [inline app/api/v2/public/markets.rb:23]: for sure remove quote_position base_position

File: app/api/v2/public/markets.rb [hunk 6]
```diff
@@ -16,9 +16,52 @@ class OrderBook < Struct.new(:asks, :bids); end
...
+                     values: { value: %w(id position quote_position base_position), message: 'public.markets.invalid_order_by' },
+                     default: 'position',
+                     desc: 'Name of the field, which result will be ordered by.'
+            optional :base_unit,
```
Comment (ysv) [inline app/api/v2/public/markets.rb:27]: add doc for difference between base_unit & search

File: app/api/v2/public/markets.rb [hunk 7]
```diff
@@ -16,9 +16,52 @@ class OrderBook < Struct.new(:asks, :bids); end
...
+              optional :base_code,
+                       type: String
+              optional :quote_code,
+                       type: String
+              optional :base_name,
+                       type: String
+              optional :quote_name,
+                       type: String
+            end
+            optional :extended,
```
Comment (ysv) [inline app/api/v2/public/markets.rb]: remove

File: app/models/market.rb [hunk 8]
```diff
@@ -49,6 +49,9 @@ class Market < ApplicationRecord
 
   # == Relationships ========================================================
 
+  has_one :base, class_name: 'Currency', foreign_key: :id, primary_key: :base_unit
```
Comment (ysv) [inline app/models/market.rb:58]: @mnaichuk decide

**Review by mnaichuk — APPROVED** [review]

**Pull request merged**